syntax = "proto2";
package DeepLearning;g
message NeuralNetParameter {
	optional string name = 1;
	optional string type = 2;
	repeated LayerStructParameter layerStruct = 100;
	optional NeuralNetTrainingParameter neuralNetTrainingParameter = 101;

}

message LayerStructParameter {
	enum ActivationType {sigmoid = 1; tanh = 2; linear = 3; softmax = 4;}
	optional int32 inputDim = 1;
	optional int32 outputDim = 2;
	optional ActivationType activationType = 3;
	optional string name = 4;
	optional string type = 5;	

}


message NeuralNetTrainingParameter {
	optional double learningRate = 1;
	optional int32 maxIter = 2;
	optional int32 miniBatchSize = 3;
	optional int32 NEpoch = 4;
	optional double epi = 5 [default = 1e-6];

}

